MidasCivil_Adapter/Convert/ToMidasCivil/Loads/FromBarDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,99 @@
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.Loads;
+
+namespace BH.Adapter.Adapters.MidasCivil
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static List <string> FromBarDifferentialTemperatureLoad(this BarDifferentialTemperatureLoad load, string assignedBar, string temperatureUnit)
+        {
+            List <string> midasBarLoad = new List<string>() ;
+
+            var barGroups = load.Objects.Elements.GroupBy(x => x.SectionProperty);
+                if (load.TemperatureProfile.Keys.Count > 20)
+                {
+                Engine.Reflection.Compute.RecordWarning("There is a maximium of 20 layers in the temperature profile");
+                    return null;
+                }
+            var barGroup = barGroups.First();
+
+            //foreach (var barGroup in barGroups)
+            //{
+                if (barGroup.First().SectionProperty == null)
+                {
+                    Engine.Reflection.Compute.RecordWarning("Section Property is required for inputting differential temperature load");
+                    return null;
+                }
+                ISectionProperty sectionProperty = barGroup.First().SectionProperty;
+
+                double presetWidth = sectionProperty.Vpy + sectionProperty.Vy / 2;
```

Comment:
See this comment where I said we should idealising the section as a rectangular section as per the guidance for in MidasCivil: https://github.com/BHoM/MidasCivil_Toolkit/pull/273#discussion_r535163618

So the width would be something like this.

```suggestion
                double presetWidth = sectionProperty.Area/(sectionProperty.Vpz + sectionPropertyVz);
```